### DISCUSSION TITLE
INAV for G4 based flight controllers

### DISCUSSION THREAD

**Discussion by velocistafpv** [iNavFlight/inav#10142] [Support]
Can INAV support or would there be future support for G4 based flight controllers?

I'm currently using several GEPRC Taker G4 AIOs for my planes and have them hooked up to DJI O3 to provide FPV with OSD. Would be nice to have INAV to show swtiches in the OSD as BF doesn't. I also have the Foxeer F745 V3 AIO which INAV supports but that is one expensive AIO.

**Comment by mmosca** [comment]
Potentially yes, but there is little incentive to add a MCU with only 512k of flash.

No one is currently working on that though.

**Reply by Wilvardson** [reply]
Hello, is there a workaround? I would be interested in using iNav on my drone that uses that flight controller.

**Reply by sensei-hacker** [reply]
You might be able to for INAV 8.
But the G4 is specifically a low-power, low-performance chip designed for things like thermostat, which needs to take a simple action once every few minutes. Not for a complex loop that runs 4,000 times per second.
So I wouldn't get it on the G4 being able to handle INAV 10 a year from now.

You could TRY adding the G4 to Cmake build configuration. There's a chance it could mostly just work, other than probably adding some defines where we already have some to distinguish between H743, F745, and F405.  If I had to guess, I'd say 20% chance it ends up being pretty easy, as programming tasks go.

More likely it would be significantly more involved. It's not particularly likely that the INAV devs who know the intricacies involved will want to take that up as project for the future because it's more interesting to make powerful new features and leverage new high performance chips to do great stuff. There just isn't much payoff in porting to smaller, lower-performance chips as a plan for the future.
